src/Guesser/ChainGuesser.php

```diff
@@ -56,7 +57,14 @@ public function guessType($object, $name, $classes)
             if ($guesser->supportObject($object)) {
                 if ($type === null) {
                     $type = $guesser->guessType($object, $name, $classes);
-                    continue;
+
+                    // DateTime guesser should not end up in multiple types
+                    // TODO: there should be a more generic solution for this
+                    if ($guesser instanceof DateTimeGuesser) {
```

Comment:
It end up being mixed only if no others type has been found and since, we exclude the the guess on the simple type with the format it should be ok IMO